Lib/test/test_asyncio/test_base_events.py

```diff
@@ -2226,6 +2226,34 @@ def test_set_nodelay(self):
             self.check_set_nodelay(sock)
 
 
+    def check_set_quickack(self, sock):
+        # quickack already true by default on some OSes
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        if opt:
+            base_events._unset_quickack(sock)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertFalse(opt)
+
+        base_events._set_quickack(sock)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertTrue(opt)
+
+    @unittest.skipUnless(hasattr(socket, 'TCP_QUICKACK'),
+                         'need socket.TCP_QUICKACK')
+    def test_set_quickack(self):
+        sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
+                             proto=socket.IPPROTO_TCP)
+        with sock:
+            self.check_set_quickack(sock)
```

Comment:
Can you use subTest to detect whether we are testing the blocking vs the non-blocking version?

Reply:
Sure, I'll update the test_set_nodelay test as well, again just for consistency (I copied and modified it).